Directories in the file.include may break the rootfs by overwriting sym links

**What happened**:

The build broke by adding files to the `file.include` directory of the GCP feature:
```
...
### gcp:
  ./usr/ ./usr/bin/ ./usr/bin/google_set_hostname ./usr/bin/google_optimize_local_ssd ./usr/bin/google_set_multiqueue ./lib/ ./lib/udev/ ./lib/udev/google_nvme_id ./lib/udev/rules.d/ ./lib/udev/rules.d/65-gce-disk-naming.rules ./lib/udev/rules.d/64-gce-disk-removal.rules ./etc/ ./etc/kernel/ ./etc/kernel/cmdline.d/ ./etc/kernel/cmdline.d/00-cmdline.cfg ./etc/dhcp/ ./etc/dhcp/dhclient.d/ ./etc/dhcp/dhclient.d/google_hostname.sh ./etc/sysctl.d/ ./etc/sysctl.d/60-gce-network-security.conf ./etc/modprobe.d/ ./etc/modprobe.d/gce-blacklist.conf ./etc/rsyslog.d/ ./etc/rsyslog.d/90-google.conf
### gardener:
  ./etc/ ./etc/modules-load.d/ ./etc/modules-load.d/ipvs.conf ./etc/docker/ ./etc/docker/daemon.json ./etc/systemd/ ./etc/systemd/system/ ./etc/systemd/system/containerd.service.d/ ./etc/systemd/system/containerd.service.d/override.conf
## executing exec.config
### base:
  chroot: failed to run command '/usr/bin/env': No such file or directory
make: *** [Makefile:67: gcp] Error 127
...
```
As one can see, the build couldn't execute the `exec.config` properly since the rootfs was broken after the file.include task has been executed.

**What you expected to happen**:

Adding simple files to `file.include` should not break the rootfs and let the build fail afterwards.

**How to reproduce it (as minimally and precisely as possible)**:

For example adding a test file into the following directory:
```
$ mkdir -p features/gcp/file.include/lib
$ touch features/gcp/file.include/lib/test
...
$ make gcp
```

**Anything else we need to know**:

The bug is caused by the `bin/garden-config` script during a Garden Linux build. In general, files located in a `file.include` directory are put into a tarball and extracted to the rootfs of the newly created Garden Linux image later on during the `copying from file.include` task (see [bin/garden-config:L47](https://github.com/gardenlinux/gardenlinux/blob/main/bin/garden-config#L47))

By extracting the tarball of the files located in the `file.include` directory to the rootfs, the used extract command does not preserve existing sym links in the rootfs of the newly created Garden Linux image. As a result, existing sym links like `lib -> /usr/lib` are overwritten by directories put into the `file.include` directory as soon as the path matches with the path of the symlink.

This can be fixed by simply adding the -h parameter to the tar command like this:
```
tar --owner=0 --group=0 -cC $featureDir/$i/file.include . | tar -xvhC $targetDir 2>&1 | grep -v "^./$" | paste -sd' '  - | indent
```

I'm not sure, if there are cases in which it is intended to overwrite the sym links with corresponding directories of the `file.include` directory but in my opinion it is safer to keep the sym links and omit creating almost empty directories. There is always the possibility to use the exec.config script to mimic this behavior if needed

**Environment**:

--- a/bin/garden-config
+++ b/bin/garden-config
@@ -44,7 +44,7 @@ printf "## copying from file.include\n"
 for i in $(tr ',' '\n' <<< $features); do
 	if [ -d "$featureDir/$i/file.include" ]; then
 		printf "### $i:\n"
-		tar --owner=0 --group=0 -cC $featureDir/$i/file.include . | tar -xvC $targetDir 2>&1 | grep -v "^./$" | paste -sd' '  - | indent
+		tar --owner=0 --group=0 -cC $featureDir/$i/file.include . | tar -xvhC $targetDir 2>&1 | grep -v "^./$" | paste -sd' '  - | indent
 	fi
 done
 

